include/dlaf/hpx_partitions.h

```diff
@@ -0,0 +1,38 @@
+
+#pragma once
+
+/// @file
+
+#include <mpi.h>
+
+#include <hpx/include/resource_partitioner.hpp>
+#include <hpx/program_options/options_description.hpp>
+
+namespace dlaf {
+
+// Note: A thread pool must be declared before starting the runtime (hpx::init())
+void try_init_mpi_pool(hpx::program_options::options_description& desc, int argc, char** argv) {
```

Comment:
Also this one may be part of the previously cited https://github.com/eth-cscs/DLA-Future/issues/100.